### PR TITLE
Repo cleanup - May

### PR DESCRIPTION
**Pull request by gewarren**
Contributes to #29115

### CODE REVIEW

**Review by tdykstra — APPROVED** [review]
LGTM

**Comment by gewarren** [comment]
@BillWagner Is this a good spot for the warning waves doc? https://review.docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/warning-waves?branch=pr-en-us-29498

Also, the AnalysisLevel property seems to be shared between C# compiler messages and Roslyn code analysis. I'll log an issue to make note of this in the docs.

File: docs/csharp/language-reference/compiler-messages/toc.yml [hunk 1]
```diff
@@ -1561,6 +1561,8 @@ items:
     href: cs8410.md
   - name: CS8411
     href: cs8411.md
+- name: Warning waves
```
Comment (BillWagner) [inline docs/csharp/language-reference/compiler-messages/toc.yml:1564]: Great catch @gewarren I can't believe I missed adding this.

This new article should replace the node "Level 5 warning messages", starting at line 1882. I'll add a second comment there.

I can open a separate PR for this if you want.

File: docs/csharp/language-reference/compiler-messages/toc.yml [hunk 2]
```diff
@@ -1882,4 +1884,4 @@ items:
 - name: Level 5 warning messages
   items:
   - name: CS8892
-    href: cs8892.md
+    href: warning-waves.md#cs8892---method-will-not-be-used-as-an-entry-point-because-a-synchronous-entry-point-method-was-found
```
Comment (BillWagner) [inline docs/csharp/language-reference/compiler-messages/toc.yml:1887]: This section should be removed and replaced with the warning waves

Line 1384 and 1385 should go here. In addition, the `displayName` metadata should list all the warning IDs in that file.

Should I open a separate PR for this?

**Review by BillWagner — COMMENTED** [review]
Thanks @gewarren 

This looks great.

I left comments around the C# compiler warning TOC. I can open a separate PR for those if you'd prefer.

**Review by BillWagner — APPROVED** [review]

File: docs/csharp/language-reference/compiler-messages/toc.yml [hunk 3]
```diff
@@ -1879,7 +1879,6 @@ items:
     href: cs1610.md
   - name: CS1712
     href: ../../misc/cs1712.md
-- name: Level 5 warning messages
-  items:
-  - name: CS8892
-    href: cs8892.md
+- name: Warning waves
```
Comment (gewarren) [inline docs/csharp/language-reference/compiler-messages/toc.yml:1882]: @BillWagner Does this look correct?

Comment (BillWagner) [reply docs/csharp/language-reference/compiler-messages/toc.yml:1882]: That's perfect! Let's :shipit:

**Pull request merged**